src/reductiveContext.re

```diff
@@ -8,35 +8,50 @@ module type Config = {
 
 module Make = (Config: Config) => {
   open Subscription;
-  let storeContext = React.createContext(Config.store);
+  let storeContext = React.createContext(None);
 
   module ContextProvider = {
     let make = React.Context.provider(storeContext);
-    let makeProps = (~value, ~children, ()) => {
+    let makeProps =
+        (
+          ~value: option(Reductive.Store.t(Config.action, Config.state)),
+          ~children,
+          (),
+        ) => {
       "value": value,
       "children": children,
     };
   };
 
   module Provider = {
     [@react.component]
-    let make = (~children) => {
-      <ContextProvider value=Config.store> children </ContextProvider>;
+    let make = (~children, ~store) => {
+      <ContextProvider value={Some(store)}> children </ContextProvider>;
```

Comment:
Does forcing a store really make sense here? I'd propose something like this, which would make this a non-breaking change and would allow for the common/simple case to work as expected without any extra work.

```reason
    let make = (~children, ~store=Config.store) => {
      <ContextProvider value=store> children </ContextProvider>;
```

I am less certain about then removing the option unwrapping code. I think going that far doesn't make sense, but it would be nice to avoid a breaking.

Reply:
Do you mean we should keep `store` passed in `Config`, and use it if no store is passed in `Provider`?
Then there are going to be two places where one could pass a store, does it make sense?

Reply:
I guess it also depends how closely we want to map this API to the react-redux one, because it's not optional in the react-redux provider